authors/apps/articles/views.py

```diff
@@ -99,3 +102,34 @@ def retrieve(self, request, slug):
         serializer = self.detailSerializer()
         serializer.instance = article_instance
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+class BookMarkArticleAPIView(generics.ListCreateAPIView):
+
+
+    renderer_classes = (ArticleJSONRenderer, )
+    queryset = Article.objects.all()
+    permission_classes = (IsAuthenticated,)
+    model = BookmarkingArticles
+
+    def post(self, request, slug):
+    
+        user_info = JWTAuthentication().authenticate(request)
```

Comment:
Since you are already including the `IsAuthenticated`  permission in your permission classes, the framework will do the JWT decoding for you, just grab the authenticated user off the request object
```python
 def post(self, request, slug):
    
        user_info = request.user
```